### DISCUSSION TITLE
Removing ui.notification() from the meta card after being notified

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#2132] [Q&A] [answered]
Hi, I'm trying to show the user logged out status after the user logs out. When I try to re-login from the same page, While creating new cards the notification is shown again. Is this a bug? Can this be avoided?

https://github.com/h2oai/wave/assets/117831190/0afbdedd-5ee3-45ee-a903-3e6dd3f157cf

**Comment by vopani** [comment]
You should set it back to `None` after notifying.

eg:

```python
q.page['meta'].notification = 'User logged out successfully...'
await q.page.save()
q.page['meta'].notification = None
await q.page.save()
```

**Reply by n-srinidhi** [reply]
@vopani Its a notification bar, so when I do this, It doesnt display at all.

```
# Show notification
    q.page['meta'].notification_bar=ui.notification_bar(
        text='User logged out successfully..',
        type='success',
        position='top-right',
        name='notif_logout',
        timeout=5,
    )
    await q.page.save()
    q.page['meta'].notification_bar=None
    await q.page.save()
```

**Reply by vopani** [reply, marked as answer]
You can either set notification_bar to `None` at the beginning of the main function (preferred, so that it goes away whenever there is a UI change) or you could set a manual sleep timer. There are other ways of handling it too (like an async subprocess to remove the notification bar).

eg:

```python
# Show notification
timeout = 5
q.page['meta'].notification_bar=ui.notification_bar(
    text='User logged out successfully..',
    type='success',
    position='top-right',
    name='notif_logout',
    timeout=timeout,
)
await q.page.save()
await q.sleep(timeout)
q.page['meta'].notification_bar=None
await q.page.save()
```

**Reply by mturoci** [reply]
All good advice from @vopani. Adding one more way of achieving this - listen for `dismissed` event and set to `None` afterwards.

https://wave.h2o.ai/docs/examples/meta-notification-bar-closable/

**Reply by n-srinidhi** [reply]
@vopani @mturoci Thank you!
